solr/core/src/java/org/apache/solr/cli/SolrCLI.java

```diff
@@ -215,18 +150,16 @@ public static void exit(int exitStatus) {
   public static void main(String[] args) throws Exception {
     final boolean hasNoCommand =
         args == null || args.length == 0 || args[0] == null || args[0].trim().length() == 0;
-    final boolean isHelpCommand =
-        !hasNoCommand && Arrays.asList("-h", "--help", "/?").contains(args[0]);
+    final boolean isHelpCommand = !hasNoCommand && Arrays.asList("-h", "--help").contains(args[0]);
 
     if (hasNoCommand || isHelpCommand) {
       printHelp();
       exit(1);
     }
 
-    if (Arrays.asList("-version", "version").contains(args[0])) {
-      // select the version tool to be run
-      CLIO.out("Deprecated operation as of 9.8.  Please use bin/solr --version.");
-      args = new String[] {"version"};
+    if (args[0].equalsIgnoreCase("version")) {
+      CLIO.out("version is not a valid command!  Did you mean --version?\n");
+      exit(1);
```

Comment:
so this is a bit of a werid one, which is right now `bin/solr version` works the same as `bin/solr --version`...    Should we keep the check and just give the same stock error message that we do if you do `bin/solr fakecommand`, which would be "fakecommand is not a valid command!"???

Reply:
I feel the `--version` flag should report the version of the local Solr CLI program, but if we implement a `version` cmd that accepts `--solr-url` then it should report the version of the remote solr host?

Reply:
I think at that point, I might suggest we introduce a `bin/solr inspect` command..   that could be then used to do lots of interesting things...    

so we don't end up with `bin/solr --version` and `bin/solr version -s http://mycorp.dev.solr:234525` ;-)

Reply:
I went ahead and took your suggestion @malliaridis

Reply:
> I think at that point, I might suggest we introduce a bin/solr inspect command.. that could be then used to do lots of interesting things...

I like this idea, especially if it comes to multiple Solr clusters / instances, this would probably be helpful. But I think this is more of a command we may introduce in the future.

If I remember correct, we also discussed the command `solr version` and decided to drop support, rather than implement functionality under this command for supporting urls as well.